### DISCUSSION TITLE
Manual installation

### DISCUSSION THREAD

**Discussion by ChenAo-Phys** [QuSpin/QuSpin#665] [General]
To make quspin compatible with the newest version of numpy, it's necessary to manually install the `dev_0.3.8` branch. However, the suggested command `python setup.py install --default-compiler-flags` doesn't always work. It's common to raise `KeyError: 'CFLAGS'`, as also mentioned in #610, #624, #650.

Marin solved this problem for me. Could you post the solution here @mgbukov ?

**Comment by mgbukov** [comment]
here it is:

1. clone the quspin repo and pull the most recent (currently dev_0.3.8) branch

2. create a new conda environment using the packages in the attached txt file as follows:

`conda create -n <environment-name> --file --quspin-linux_reqs.txt`

3. activate this environment 

4. enter the directory hosting the most recent development branch files (it should contain the setup.py file) and run:

`python setup.py install build_ext -i --omp`

It'll take some time; while compiling it'll show a bunch of warning harmless messages; if it compiles without an error, you can go into \tests and try a few tests to see if the installation was successful.

If the manually installed quspin raises a circular import error, you ay copy the quspin source code into conda site-packages.

[quspin-linux_reqs.txt](https://github.com/QuSpin/QuSpin/files/14397736/quspin-linux_reqs.txt)

**Comment by ChenAo-Phys** [comment]
Just realized a minor mistake. The command for creating new environment should be
`conda create -n <environment-name> --file quspin-linux_reqs.txt`
